plugins/webapi/ledgerstate/output.go

```diff
@@ -0,0 +1,168 @@
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, NewErrorResponse(err))
+	}
+
+	cachedConsumers := messagelayer.Tangle().LedgerState.Consumers(outputID)
+	defer cachedConsumers.Release()
+
+	consumers := cachedConsumers.Unwrap()
+	// check all returned and unwrapped objects
+	for _, consumer := range consumers {
+		if consumer == nil {
+			return c.JSON(http.StatusBadRequest, NewErrorResponse(xerrors.Errorf("failed to load consumers")))
+		}
+	}
+
+	return c.JSON(http.StatusOK, NewConsumers(consumers))
+}
+
+// GetOutputMetadataEndPoint is the handler for the /ledgerstate/outputs/:outputID/metadata endpoint.
+func GetOutputMetadataEndPoint(c echo.Context) (err error) {
+	outputID, err := ledgerstate.OutputIDFromBase58(c.Param("outputID"))
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, NewErrorResponse(err))
+	}
+
+	cachedOutputMetadata := messagelayer.Tangle().LedgerState.OutputMetadata(outputID)
+	defer cachedOutputMetadata.Release()
+	// outputMetadata object is nil if was not found
+	outputMetadata := cachedOutputMetadata.Unwrap()
+	if outputMetadata == nil {
+		return c.JSON(http.StatusBadRequest, NewErrorResponse(xerrors.Errorf("output metadata not present in node")))
+	}
+
+	return c.JSON(http.StatusOK, NewOutputMetadata(outputMetadata))
+}
+
+// endregion ///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// region Output ///////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// Output represents a JSON model of a ledgerstate.Output.
+type Output struct {
+	ID       string           `json:"id"`
+	Type     string           `json:"type"`
+	Balances []ColoredBalance `json:"balances"`
+	Address  string           `json:"address"`
+}
+
+// ColoredBalance represents a JSON model of a single colored balance.
+type ColoredBalance struct {
+	Color   string `json:"color"`
+	Balance uint64 `json:"balance"`
+}
+
+// NewOutput creates a JSON compatible representation of the output.
+func NewOutput(output ledgerstate.Output) Output {
+	return Output{
+		ID:   output.ID().Base58(),
+		Type: output.Type().String(),
+		Balances: func() []ColoredBalance {
+			coloredBalances := make([]ColoredBalance, 0)
+			output.Balances().ForEach(func(color ledgerstate.Color, balance uint64) bool {
+				coloredBalances = append(coloredBalances, ColoredBalance{Color: color.String(), Balance: balance})
+				return true
+			})
+			return coloredBalances
+		}(),
+		Address: output.Address().Base58(),
+	}
+}
+
+// endregion ///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// region Consumers ////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// Consumers is the JSON model of consumers of the output.
+type Consumers struct {
+	ConsumerCount        int                   `json:"consumerCount"`
```

Comment:
I am not sure if we really need the ConsumerCount here as you can directly derive that from the list of Transactions but maybe it would be nice to have the OutputID named here so the response says which Consumers we are showing.

Reply:
I think it doesn't hurt to have `consumerCount`, especially when we will be using this for debugging.
I've added the `outputID`.